security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisProvider.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security;
+
+import com.powsybl.commons.Versionable;
+import com.powsybl.commons.config.PlatformConfigNamedProvider;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.contingency.ContingenciesProvider;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.security.interceptors.SecurityAnalysisInterceptor;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Security analysis provider
+ *
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public interface SecurityAnalysisProvider extends Versionable, PlatformConfigNamedProvider {
+
+    /**
+     * Run an asynchronous single security analysis job.
+     *
+     * @param network IIDM network on which the security analysis will be performed
+     * @param workingVariantId network variant ID on which the analysis will be performed
+     * @param detector
+     * @param filter
+     * @param computationManager
+     * @param parameters specific security analysis parameters
+     * @param contingenciesProvider provides list of contingencies
+     * @param interceptors
+     * @return a {@link CompletableFuture} on {@link SecurityAnalysisResult} that gathers sensitivity factor values
+     */
+    CompletableFuture<SecurityAnalysisResult> run(Network network,
+                                                  String workingVariantId,
+                                                  LimitViolationDetector detector,
```

Comment:
For me LimitViolationDetector has nothing to do in the API, because it supposes that violation detection is done on IIDM model. It is not the case for instance in OLF, because violation detection is  done one a more internal data model to speed up performance. I guess it should be the case too, with hades2.
@sylvlecl i think this is you who add it to the API, do you really need it?

Reply:
Indeed it's an important component for us, to allow custom violation detection, for example taking into account limit reduction factors based on the branch, the contingency, or the limit itself (acceptable duration ... ).

A goal of the detector was indeed to try to avoid differences in the way violations are created by different implementations, and to provide the same features to different implementations (like this "limit reduction factors" feature).

If we don't want to use it here anymore, this kind of processing will need to be done "externally" by a calling process, in this case :
* we should be very clear in the javadoc how detection violation should be performed by the implementation (for example, should it provide 1 violation for each exceeded limit, or only 1 for the "worse")
* a detail to be solved is the fact that for voltage violations, for the moment, we lose the "bus" information
* `itools security-analysis` will lose some functionality, except if we add the possibility to plug in some post-processing

Reply:
Issue #1602 was created to keep these comments